include/deal.II/dofs/dof_handler.h

```diff
@@ -1224,6 +1243,33 @@ DoFHandler<dim,spacedim>::get_fe () const
 
 template <int dim, int spacedim>
 inline
+const FiniteElement<dim,spacedim> &
+DoFHandler<dim,spacedim>::get_finite_element
+(const unsigned int index) const
+{
+  (void) index;
+  Assert(index == 0, ExcMessage("There is only one FiniteElement stored. The index must be zero!"));
+  Assert(selected_fe!=nullptr,
+         ExcMessage("You are trying to access the DoFHandler's FiniteElement object before it has been initialized."));
+  return *selected_fe;
+}
+
+
+
+template <int dim, int spacedim>
+inline
+hp::FECollection<dim,spacedim>
+DoFHandler<dim,spacedim>::get_fe_collection () const
+{
+  Assert(selected_fe!=nullptr,
+         ExcMessage("You are trying to access the DoFHandler's FiniteElement object before it has been initialized."));
+  return hp::FECollection<dim,spacedim>(*selected_fe);
```

Comment:
To name just one example: a call like
```
  hp::FEValues<dim> x_fe_values (dof_handler.get_fe_collection(), ...);
```
is going to ask for certain trouble if you return a temporary here.